schemas/kb.py

```diff
@@ -0,0 +1,81 @@
+"""
+Knowledge Base Schema
+
+This is a very general schema that covers many information extraction-style
+tasks, including:
+
+- Named entity recognition (NER)
+- Named entity disambiguation/canonicalization/normalization (NED)
+- Event extraction
+- Relation extraction (RE)
+- Coreference resolution
+
+This schema assumes a document with child elements (e.g., entities, relations)
+and a flat hierarchy of document passages.
+
+"""
+import datasets
+
+
+features = datasets.Features(
+    {
+        "id": datasets.Value("string"),
+        "document_id": datasets.Value("string"),
```

Comment:
I think it would be nice to have a top level `text` field here that any offset would work in ... for example
```
text = sample["text"]
ii = sample["passages"][n]["offsets"][0][0]
ff = sample["passages"][n]["offsets"][0][1]
assert(text[ii:ff] = sample["passages"][n]["text"])
# same for entities 
```

Otherwise, if I'm looking for the context of an entity I have to search through each passage to discover if the entity offset is within the offset range of that passage. This could be as straightforward as, 
```
sample["text"] = " ".join([passage['text'] for passage in passages])
```

It definitely leads to some duplicate information but I think its worth it for reasonable sized datasets ... it would also give the unit tests something to check.

Reply:
coming back to this (sorry for the long comments!) ... I now see that my suggestion would be particularly silly for samples with one passage (then the top level text would just be an exact copy of the `passages[0]['text']`  ... I withdraw my top level text suggestion :)